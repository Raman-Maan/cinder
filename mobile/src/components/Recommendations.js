import React from 'react';
import { StyleSheet, Image, ScrollView, TouchableOpacity, Animated, Easing } from 'react-native';
import { Container, View, DeckSwiper, Card, CardItem, H3, Text, Left, Body, Icon, Button } from 'native-base';

import cinder from '../theme/variables/cinder';

const cards = [
  {
    text: 'Card Uno',
    name: 'Mac Miller',
    age: 21,
    image: 'https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b',
  }, {
    text: 'Card Two',
    name: 'Kendrick Lamar',
    age: 21,
    image: 'http://cache.umusic.com/_sites/kendricklamar.com/images/og.jpg',
  }, {
    text: 'Card Three',
    name: 'SZA',
    age: 21,
    image: 'https://media.pitchfork.com/photos/59298fe813d1975652136c25/1:1/w_300/05bc322d.jpg',
  },
];

const styles = StyleSheet.create({
  container__view: {
    flex: 1,
  },
  recommends__card: {
    overflow: 'hidden',
    backgroundColor: 'transparent',
  },
  recommends_details: {
    flexDirection: 'row',
    flex: 1,
    position: 'absolute',
    top: 459,
    left: 0,
    right: 0,
    justifyContent: 'space-between',
    padding: 10,
    backgroundColor: 'rgba(0, 0, 0, 0.3)',
    borderBottomLeftRadius: cinder.cardBorderRadius,
    borderBottomRightRadius: cinder.cardBorderRadius,
  },
  recommends__details_text: {
    color: '#fff',
  },
  recommends__about: {
    maxHeight: 400,
    flexDirection: 'column',
    backgroundColor: 'white',
  },
  recommends__image: {
    height: 500,
    flex: 1,
    borderRadius: cinder.cardBorderRadius,
  },
  buttons__container: {
    flexDirection: 'row',
    flex: 1,
    position: 'absolute',
    bottom: 30,
    left: 0,
    right: 0,
    justifyContent: 'space-between',
    padding: 15,
  },
});

export default class Recommendations extends React.Component {
  constructor(props) {
    super(props);

    this.setSwiper = this.setSwiper.bind(this);
    this.setScrollView = this.setScrollView.bind(this);
    this.renderEmpty = this.renderEmpty.bind(this);
    this.renderRecommendation = this.renderRecommendation.bind(this);
    this.renderButtons = this.renderButtons.bind(this);

    this.swiper = null;
    this.scrollView = null;

    this.animatedValue = new Animated.Value(0);
  }

  componentDidMount() {
    this.animate();
  }

  setScrollView(ref) {
    this.scrollView = ref;
  }

  setSwiper(ref) {
    this.swiper = ref;
  }

  animate() {
    this.animatedValue.setValue(0);
    Animated.sequence([
      Animated.timing(this.animatedValue, {
        toValue: 1,
        duration: 1000,
        easing: Easing.ease,
      }),
      Animated.timing(this.animatedValue, {
        toValue: 0,
        duration: 1000,
        easing: Easing.ease,
      }),
    ]).start(() => this.animate());
  }


  renderEmpty() {
    return (
      <View style={{ alignSelf: 'center' }}>
        <Text>No recommendations left! :(</Text>
      </View>
    );
  }

  renderRecommendation(item) {
    const bounce = this.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: [0, 3],
    });

    const scale = this.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: [0.9, 1],
    });

    return (
      <Card style={styles.recommends__card}>
        <ScrollView ref={this.setScrollView} style={{ flex: 1, maxHeight: 500, flexGrow: 0 }}>
          <Image style={styles.recommends__image} source={{ uri: item.image }} />
          <View style={styles.recommends_details}>
            <View style={{ flexDirection: 'row' }}>
              <Text bold style={styles.recommends__details_text}>{item.name}</Text>
              <Text style={styles.recommends__details_text}>{`, ${item.age}`}</Text>
            </View>
            <Animated.View style={{ transform: [{ translateY: bounce }, { scale }] }}>
              <TouchableOpacity onPress={() => this.scrollView.scrollToEnd()}>
                <Icon name="ios-arrow-down" style={{ color: 'white', padding: 0, fontSize: 24 }} />
              </TouchableOpacity>
            </Animated.View>
          </View>
          <CardItem details style={styles.recommends__about}>
            <H3>{`About ${item.name.split(' ')[0]}`}</H3>
            <Text left>{item.name}</Text>
          </CardItem>
        </ScrollView>
      </Card>
    );
  }

  renderButtons() {
    return (
      <View style={styles.buttons__container}>
        <Button danger rounded bordered iconLeft onPress={() => this.swiper._root.swipeLeft()}>
          <Icon name="close" />
          <Text>Pass</Text>
        </Button>
        <Button primary rounded iconRight onPress={() => this.swiper._root.swipeRight()}>
          <Text>Like</Text>
          <Icon name="heart" />
        </Button>
      </View>
    );
  }

  render() {
    return (
      <Container>
        <View style={styles.container__view}>
          <DeckSwiper
            ref={this.setSwiper}
            looping // TODO - set to false, only for testing
            dataSource={cards}
            renderEmpty={this.renderEmpty}
            renderItem={this.renderRecommendation}
          />
        </View>
        {this.renderButtons()}
      </Container>
    );
  }
}
