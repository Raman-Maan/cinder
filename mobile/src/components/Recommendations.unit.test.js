import React from 'react';
import renderer from 'react-test-renderer';

import Recommendations from './Recommendations';

const testData = [
  {
    userID: 1,
    userName: 'Mac Miller',
    age: 21,
    primaryPic: 'https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b',
    userBio: 'He raps, he does some other stuff too but mostly just that',
  }, {
    userID: 2,
    userName: 'Kendrick Lamar',
    age: 21,
    primaryPic: 'http://cache.umusic.com/_sites/kendricklamar.com/images/og.jpg',
    userBio: 'Bad dancer for real',
  }, {
    userID: 3,
    userName: 'SZA',
    age: 21,
    primaryPic: 'https://media.pitchfork.com/photos/59298fe813d1975652136c25/1:1/w_300/05bc322d.jpg',
    userBio: 'She is very hot',
  },
];

it('renders when empty', () => {
  const rendered = renderer.create(<Recommendations data={[]} />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('renders with cards', () => {
  const rendered = renderer.create(<Recommendations data={testData} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
