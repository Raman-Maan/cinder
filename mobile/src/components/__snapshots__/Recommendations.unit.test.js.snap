// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recommendations /> renders with data 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "position": "relative",
        }
      }
    >
      <View>
        <View
          collapsable={undefined}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "left": 0,
              "opacity": 0.8,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "scale": 0.8,
                },
              ],
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#ccc",
                  "borderLeftWidth": 0.5,
                  "borderRadius": 2,
                  "borderRightWidth": 0.5,
                  "borderTopWidth": 0.5,
                  "elevation": 3,
                  "flex": 1,
                  "flexWrap": "nowrap",
                  "marginBottom": 5,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 5,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 1.5,
                },
                Object {
                  "backgroundColor": "transparent",
                  "overflow": "hidden",
                },
              ]
            }
            virtual={undefined}
          >
            <RCTScrollView
              style={
                Object {
                  "flex": 1,
                  "flexGrow": 0,
                  "maxHeight": 500,
                }
              }
            >
              <View>
                <Image
                  source={
                    Object {
                      "uri": "http://cache.umusic.com/_sites/kendricklamar.com/images/og.jpg",
                    }
                  }
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "flex": 1,
                      "height": 500,
                      "resizeMode": "contain",
                      "width": null,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.3)",
                        "borderBottomLeftRadius": 20,
                        "borderBottomRightRadius": 20,
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "left": 0,
                        "padding": 10,
                        "position": "absolute",
                        "right": 0,
                        "top": 456,
                      },
                    ]
                  }
                  virtual={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {},
                        Object {
                          "flexDirection": "row",
                        },
                      ]
                    }
                    virtual={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      bold={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#fff",
                          },
                        ]
                      }
                      uppercase={false}
                      virtual={undefined}
                    >
                      Kendrick Lamar
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#fff",
                          },
                        ]
                      }
                      uppercase={false}
                      virtual={undefined}
                    >
                      , 21
                    </Text>
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 0.9,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": undefined,
                              "fontSize": 12,
                            },
                            Array [
                              Object {
                                "color": "#000",
                                "fontSize": 30,
                              },
                              Object {
                                "color": "#000",
                                "fontSize": 30,
                              },
                              Object {
                                "color": "white",
                                "fontSize": 24,
                                "padding": 0,
                              },
                            ],
                            Object {
                              "fontFamily": "Ionicons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                        virtual={undefined}
                      >
                        Ôèê
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  details={true}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#fff",
                        "borderRadius": 2,
                        "flexDirection": "row",
                        "paddingBottom": 10,
                        "paddingLeft": 15,
                        "paddingRight": 15,
                        "paddingTop": 10,
                      },
                      Object {
                        "backgroundColor": "white",
                        "flexDirection": "column",
                        "maxHeight": 400,
                      },
                    ]
                  }
                  virtual={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#000",
                        "fontSize": 21,
                        "lineHeight": 22,
                      }
                    }
                    virtual={undefined}
                  >
                    About Kendrick
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    left={true}
                    style={
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      }
                    }
                    uppercase={false}
                    virtual={undefined}
                  >
                    Bad dancer for real
                  </Text>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
        <View
          collapsable={undefined}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#ccc",
                  "borderLeftWidth": 0.5,
                  "borderRadius": 2,
                  "borderRightWidth": 0.5,
                  "borderTopWidth": 0.5,
                  "elevation": 3,
                  "flex": 1,
                  "flexWrap": "nowrap",
                  "marginBottom": 5,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 5,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 1.5,
                },
                Object {
                  "backgroundColor": "transparent",
                  "overflow": "hidden",
                },
              ]
            }
            virtual={undefined}
          >
            <RCTScrollView
              style={
                Object {
                  "flex": 1,
                  "flexGrow": 0,
                  "maxHeight": 500,
                }
              }
            >
              <View>
                <Image
                  source={
                    Object {
                      "uri": "https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b",
                    }
                  }
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "flex": 1,
                      "height": 500,
                      "resizeMode": "contain",
                      "width": null,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.3)",
                        "borderBottomLeftRadius": 20,
                        "borderBottomRightRadius": 20,
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "left": 0,
                        "padding": 10,
                        "position": "absolute",
                        "right": 0,
                        "top": 456,
                      },
                    ]
                  }
                  virtual={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {},
                        Object {
                          "flexDirection": "row",
                        },
                      ]
                    }
                    virtual={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      bold={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#fff",
                          },
                        ]
                      }
                      uppercase={false}
                      virtual={undefined}
                    >
                      Mac Miller
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#fff",
                          },
                        ]
                      }
                      uppercase={false}
                      virtual={undefined}
                    >
                      , 21
                    </Text>
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "transform": Array [
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 0.9,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": undefined,
                              "fontSize": 12,
                            },
                            Array [
                              Object {
                                "color": "#000",
                                "fontSize": 30,
                              },
                              Object {
                                "color": "#000",
                                "fontSize": 30,
                              },
                              Object {
                                "color": "white",
                                "fontSize": 24,
                                "padding": 0,
                              },
                            ],
                            Object {
                              "fontFamily": "Ionicons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                        virtual={undefined}
                      >
                        Ôèê
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  details={true}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#fff",
                        "borderRadius": 2,
                        "flexDirection": "row",
                        "paddingBottom": 10,
                        "paddingLeft": 15,
                        "paddingRight": 15,
                        "paddingTop": 10,
                      },
                      Object {
                        "backgroundColor": "white",
                        "flexDirection": "column",
                        "maxHeight": 400,
                      },
                    ]
                  }
                  virtual={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#000",
                        "fontSize": 21,
                        "lineHeight": 22,
                      }
                    }
                    virtual={undefined}
                  >
                    About Mac
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    left={true}
                    style={
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      }
                    }
                    uppercase={false}
                    virtual={undefined}
                  >
                    He raps, he does some other stuff too but mostly just that
                  </Text>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "bottom": 30,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "left": 0,
          "padding": 15,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#d9534f",
          "borderLeftWidth": 1,
          "borderRadius": 57,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "elevation": null,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#d9534f",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 0,
                "paddingTop": 2,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        ÔêÜ
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#d9534f",
            "fontFamily": "System",
            "fontSize": 16.5,
            "lineHeight": 19,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        Pass
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 57,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "lineHeight": 19,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        Like
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#fff",
                "fontSize": 24,
                "marginLeft": 0,
                "marginRight": 16,
                "paddingTop": 2,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        ÔëÇ
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Recommendations /> should render when empty 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "position": "relative",
        }
      }
    >
      <View>
        <View
          style={
            Array [
              Object {},
              Object {
                "alignSelf": "center",
              },
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            No recommendations left! :(
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
